open-craft/opencraft
# [BB-3618] Implement page structure

We are trying to migrate to the 2 columns new page structure for the console. This way the UI has an overlay on the sidebar nav to make all the adjustments possible.

**JIRA tickets**: [BB-3618](https://tasks.opencraft.com/browse/BB-3618)

**Discussions**: [Public GitLab Ticket](https://gitlab.com/opencraft/dev/opencraft/-/issues/701)

**Dependencies**:  ~~This is still dependent on https://github.com/open-craft/opencraft/pull/701 but initially set of reviews can be done irrespective. Later we can integrate the studio page.~~

**Screenshots**: 
![image](https://user-images.githubusercontent.com/7670449/108347293-5bc5ec00-7206-11eb-8d08-e155ceeab919.png)

![image](https://user-images.githubusercontent.com/7670449/108347435-7e580500-7206-11eb-8d72-67fb38df5d2d.png)

![image](https://user-images.githubusercontent.com/7670449/109959401-6d7eb780-7d0d-11eb-8db4-d64de542e437.png)




~~**Sandbox URL**: TBD - sandbox is being provisioned.~~

~~**Merge deadline**: "None" if there's no rush, "ASAP" if it's critical, or provide a specific date if there is one.~~

**Testing instructions**:

1. Install and run the OCIM interface
2. You need to comment the following lines to bypass validation
```
diff --git a/registration/api/v2/serializers.py b/registration/api/v2/serializers.py
index 5e1f28d4..08ee02fa 100644
--- a/registration/api/v2/serializers.py
+++ b/registration/api/v2/serializers.py
@@ -362,8 +362,8 @@ class OpenEdXInstanceConfigSerializer(serializers.ModelSerializer):
         is_new_instance = self.instance is None
         is_changed = not is_new_instance and self.instance.subdomain != self.initial_data.get("subdomain")
 
-        if is_new_instance or is_changed:
-            validate_available_subdomain(value)
+        # if is_new_instance or is_changed:
+        #     validate_available_subdomain(value)
 
         return value

```
3. You need to manually visit the link http://localhost:3000/newconsole
4. Click on the Logo Suboption in Accordion you will see the layout change
5.  For testing logo update page you need to revert few changes from https://github.com/open-craft/opencraft/pull/714/commits/fe0a0034d9e79c9369a32d215a094534cc5b6a41
6. The logo update page is dummy and it's just to have a concept implementation here/
7. We need to check the transition is happening when we click Course (Studio) you will be taken to new dashboard and any other option will take you to old board. 

**Author notes and concerns**:

1. We don't need to verify the LOGO component since it won't be used as-is. The point of integrating that component was to check and build the functionality for the same.
2. As mentioned when the dependent PR will get merge there will be a link from the old console page to a new one, this is done temporarily till the whole page gets created. We want to ship new features every sprint.
3. The whole point of the PR is to get the structure ready and integrate the Course (Studio) page in the new console. This way we will have the foundation of future work.

**Reviewers**
